articles/rules/preview/index.md

```diff
@@ -0,0 +1,314 @@
+* Enable counters or persist other information. (For information on storing user data, see: [Metadata in Rules](/rules/metadata-in-rules).)
+* Enable __multifactor__ authentication, based on context (e.g. last login, IP address of the user, location, etc.).
+* Modify tokens: Change the returned __scopes__ of the `access_token` and/or add claims to it, and to the `id_token`. At the moment you can do this for the [Password Exchange](/api-auth/grant/password#rules).
+
+## Video: Using Rules
+Watch this video learn all about rules in just a few minutes.
+
+<%= include('../../videos/_video', { id: 'g7dy1fpwc3' }) %>
+
+## Rule Syntax
+
+A Rule is a function with the following arguments:
+
+* `user`: the user object as it comes from the identity provider (For a complete list of the user properties, see: [User Profile Structure](/user-profile/user-profile-structure)).
+
+* `context`: an object containing contextual information of the current authentication transaction, such as user's IP address, application, location. (A complete list of context properties is available here: [Context Argument Properties in Rules](/rules/context).)
+* `callback`: a function to send back the potentially modified `user` and `context` objects back to Auth0 (or an error).
+
+<div class="alert alert-info">
+Because of the async nature of Node.js, it is important to always call the <code>callback</code> function, or else the script will timeout.
+</div>
+
+## Examples
+
+<div class="alert alert-info">
+You can find more examples of common Rules on Github at <a href="https://github.com/auth0/rules">auth0/rules</a>.
+</div>
+
+To create a Rule, or try the examples below, go to [New Rule](${manage_url}/#/rules/create) in the Rule Editor on the dashboard.
+
+### *Hello World*
+
+This rule will add a `hello` attribute to all users authenticating through any provider.
+
+```js
+function (user, context, callback) {
+  user.hello = 'world';
+  console.log('===> set "hello" for ' + user.name);
+  callback(null, user, context);
+}
+```
+
+<div class="alert alert-info">
+  You can add <code>console.log</code> lines for <a href="#debugging">debugging</a> or use the <a href="/extensions/realtime-webtask-logs">Real-time Webtask Logs Extension</a>.
+</div>
+
+
+### Add roles to a user
+
+In this example, all authenticated users will get a **guest** role, but `johnfoo@gmail.com` will also be an **admin**:
+
+```js
+function (user, context, callback) {
+  user.roles = [];
+  // only johnfoo is admin
+  if (user.email === 'johnfoo@gmail.com') {
+    user.roles.push('admin');
+  }
+
+  // all users are guest
+  user.roles.push('guest');
+
+  callback(null, user, context);
+}
+```
+
+At the beginning of the rules pipeline, John's `user` object will be:
+
+```js
+{
+  "email": "johnfoo@gmail.com",
+  "family_name": "Foo",
+  "user_id": "google-oauth2|103547991597142817347"
+  //... other properties ...
+}
+```
+
+The `context` object will be:
```

Comment:
context may include `idToken` and `accessToken` properties. this is important because now a lot of the old behavior is now handled by these two properties.